api/src/main/java/org/pmiops/workbench/db/dao/BillingProjectBufferEntryDao.java

```diff
@@ -28,18 +28,27 @@
   List<DbBillingProjectBufferEntry> findAllByStatusAndLastStatusChangedTimeLessThan(
       short status, Timestamp timestamp);
 
-  DbBillingProjectBufferEntry findFirstByStatusOrderByLastSyncRequestTimeAsc(short status);
-
   List<DbBillingProjectBufferEntry> findTop5ByStatusOrderByLastSyncRequestTimeAsc(short status);
 
   DbBillingProjectBufferEntry findFirstByStatusOrderByCreationTimeAsc(short status);
 
   Long countByStatus(short status);
 
   default Map<BufferEntryStatus, Long> getCountByStatusMap() {
-    return DaoUtils.getAttributeToCountMap(findAll(), DbBillingProjectBufferEntry::getStatusEnum);
+    return computeProjectCountByStatus().stream()
+        .collect(
+            ImmutableMap.toImmutableMap(
+                StatusToCountResult::getStatusEnum, StatusToCountResult::getNumProjects));
   }
 
+  @Query(
+      value =
+          "select status, count(billing_project_buffer_entry_id) as numpNrojects\n"
+              + "    from DbBillingProjectBufferEntry \n"
+              + "group by status\n"
+              + "order by status")
+  List<StatusToCountResult> computeProjectCountByStatus();
```

Comment:
Can we write a test class for this in `BillingProjectBufferEntryDaoTest` and test all the methods?